roiextractors/example_datasets/toy_example.py

```diff
@@ -1 +1,150 @@
-import numpy as np
\ No newline at end of file
+import numpy as np
+import spikeextractors as se
+from ..extractors.numpyextractors import NumpyImagingExtractor, NumpySegmentationExtractor
+
+
+def _gaussian(x, mu, sigma):
+    return 1 / np.sqrt(2 * np.pi * sigma) * np.exp(- (x - mu) ** 2 / sigma)
+
+
+def _generate_rois(num_units=10, size_x=100, size_y=100, roi_size=4, min_dist=5, mode='uniform'):
+    image = np.zeros((size_x, size_y))
+    max_iter = 1000
+
+    count = 0
+    it = 0
+    means = []
+
+    while count < num_units:
+        mean_x = np.random.randint(0, size_x - 1)
+        mean_y = np.random.randint(0, size_y - 1)
+
+        mean_ = np.array([mean_x, mean_y])
+
+        if len(means) == 0:
+            means.append(mean_)
+            count += 1
+        else:
+            dists = np.array([np.linalg.norm(mean_ - m) for m in means])
+
+            if np.all(dists > min_dist):
+                means.append(mean_)
+                count += 1
+
+        it += 1
+
+        if it >= max_iter:
+            raise Exception("Could not fit ROIs given 'min_dist'")
+
+    roi_pixels = []
+
+    for m, mean in enumerate(means):
+        # print(f"ROI {m + 1}/{num_units}")
+        pixels = []
+        for i in np.arange(size_x):
+            for j in np.arange(size_y):
+                p = np.array([i, j])
+
+                if np.linalg.norm(p - mean) < roi_size:
+                    pixels.append(p)
+                    if mode == 'uniform':
+                        image[i, j] = 1
+                    elif mode == 'gaussian':
+                        image[i, j] = _gaussian(i, mean[0], roi_size) + _gaussian(j, mean[1], roi_size)
+                    else:
+                        raise Exception("'mode' can be 'uniform' or 'gaussian'")
+        roi_pixels.append(np.array(pixels))
```

Comment:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.gaussian_filter.html might be cleaner

Reply:
or at least vectorizing so it's not on a for loop

Reply:
@alejoe91

Reply:
@bendichter the gaussian filter is different, as it convolves the entire image with a gaussian kernel. 
Since each gaussian have different mean, I think we can't avoid a loop here (it's very fast anyways :))